tests/test_pattern.py

```diff
@@ -60,7 +60,6 @@ def test_cmp_hash():
     pattern = context.get_source()
     other = context.get_source()
     assert pattern == other
-    assert not pattern != other
```

Comment:
Isn't that all removing tests for `__ne__` etc?

Reply:
I reverted removing the not equal tests.